### PR TITLE
Refactoring is all you need

### PR DESCRIPTION
**Pull request by fdamken**
Includes #96, #99, and #100.

### CODE REVIEW

File: Pyrado/pyrado/plotting/heatmap.py [hunk 1]
```diff
@@ -191,10 +190,6 @@ def draw_heatmap(
     :return: handles to the heat map and the color bar figures (`None` if not existent)
     """
     if isinstance(data, pd.DataFrame):
-        if not isinstance(data.index, NumericIndex):
```
[the remark this answers is not in the document]
Comment (fdamken) [reply Pyrado/pyrado/plotting/heatmap.py:194]: `NumericIndex` does not exist anymore in the latest Pandas version, it was removed in Pandas 2.0 (and I was too lazy to figure out what the new method would be).

[the remark this answers is not in the document]
Comment (famura) [reply Pyrado/pyrado/plotting/heatmap.py:194]: How about `data.index.is_numeric()`?
https://pandas.pydata.org/docs/reference/api/pandas.Index.is_numeric.html

If that doesn't work out of the box, just let me know and then we go without the checks

File: Pyrado/pyrado/domain_randomization/domain_parameter.py [hunk 2]
```diff
@@ -393,6 +393,51 @@ def make_broadening(
...
+        return SelfPacedDomainParam(
+            name=name,
+            target_mean=to.tensor(target_mean),
+            target_cov_flat=to.tensor(target_cov_flat),
+            init_mean=to.tensor(init_mean),
+            init_cov_flat=to.tensor(init_cov_flat),
+            clip_lo=-pyrado.inf,
+            clip_up=+pyrado.inf,
+        )
+
```
Comment (famura) [inline Pyrado/pyrado/domain_randomization/domain_parameter.py:440]: could you please add a test for that?

**Review by famura — COMMENTED** [review]
Sorry for being annoying about the tests midway of this PR. I just saw that the coverage of the patch  is really low, and I think that we have no (dedicated) test for `SelfPacedDomainParam`. Would you please add some? It doesn't need to be anything fancy

**Review by famura — APPROVED** [review]
Thank you very much for addressing all my comments

**Comment by famura** [comment]
@fdamken I font really understand wh the `Run Tests` action failed with
 Error: The process '/home/runner/work/SimuRLacra/SimuRLacra/Pyrado/cc-reporter' failed with exit code 255
 
 I re-ran it now.

**Comment by fdamken** [comment]
> Still the same error: https://github.com/famura/SimuRLacra/actions/runs/6606008836/job/17950282655?pr=98#step:6:795
> 
> Any ideas @miterion @fdamken ?

It's probably because my PR comes from another repo, so it should be fine. Only the reporting fails, the test themselves pass.

**Pull request merged**